### ISSUE TITLE
Segfault when closing Godot

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#5425]
**Operating system or device - Godot version:**
Mageia 6 x86_64, current HEAD (544b196)

**Issue description** (what happened, and what was expected):
Godot segfaults when it's closed.
```
Thread 5 "godot-git" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffe8a55700 (LWP 3367)]
0x0000000001668db6 in ScriptServer::thread_exit () at core/script_language.cpp:115
115   _languages[i]->thread_exit();

(gdb) bt
#0  0x0000000001668db6 in ScriptServer::thread_exit () at core/script_language.cpp:115
#1  0x000000000053a8a9 in ThreadPosix::thread_callback (userdata=0x328dd00) at drivers/unix/thread_posix.cpp:59
#2  0x00007ffff5c9e66d in start_thread () from /lib64/libpthread.so.0
#3  0x00007ffff4d27d8d in clone () from /lib64/libc.so.6
```

**Steps to reproduce:**
1. Start the project manager
2. Close it

**Comment by akien-mga** [comment]
@reduz Looks like a bug from 8dac3bf.

**Comment by akien-mga** [comment]
Fixed by 3813160
